packages/validator/src/services/attestation.ts

```diff
@@ -148,13 +154,25 @@ export class AttestationService {
       })
     );
 
-    this.metrics?.attesterStepCallPublishAttestation.observe(this.clock.secFromSlot(slot + 1 / 3));
-
     // Step 2. Publish all `Attestations` in one go
     if (signedAttestations.length > 0) {
       try {
+        const msToOneThirdSlot = this.clock.msToSlot(slot + 1 / 3);
+        const signedAttestationTime = Date.now();
+        // as monitored in hetzner nodes, most of the time we're before 1/3 of slot at this time
+        // i.e. msToOneThirdSlot > 0
+        if (msToOneThirdSlot > 0) {
+          // for early attestations, wait until now + 2000ms, or 1/3 of slot, whichever comes first
+          await sleep(Math.min(msToOneThirdSlot, this.earlyAttestationDelayMs));
+        }
+        this.metrics?.attesterStepCallPublishAttestation.observe(this.clock.secFromSlot(slot + 1 / 3));
```

Comment:
if its `this.earlyAttestationDelayMs` that comes first, won't this metric be wrong?